rotkehlchen/data_migrations/migrations/migrations_15.py

```diff
@@ -0,0 +1,53 @@
+import logging
+from typing import TYPE_CHECKING
+
+from rotkehlchen.assets.asset import EvmToken
+from rotkehlchen.chain.evm.decoding.hop.constants import CPT_HOP
+from rotkehlchen.errors.asset import UnknownAsset, WrongAssetType
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter, enter_exit_debug_log
+
+if TYPE_CHECKING:
+    from rotkehlchen.data_migrations.progress import MigrationProgressHandler
+    from rotkehlchen.rotkehlchen import Rotkehlchen
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+@enter_exit_debug_log()
+def data_migration_15(rotki: 'Rotkehlchen', progress_handler: 'MigrationProgressHandler') -> None:
+    """
+    Introduced at v1.34.1
+
+    Sets the protocol for all Hop LP tokens received in an Add liquidity event
+    """
+    # find one transaction for each Hop LP token received in an Add liquidity event
+    with rotki.data.db.conn.read_ctx() as cursor:
+        hop_lp_assets = cursor.execute(
+            'SELECT DISTINCT asset FROM evm_events_info AS EE '
+            'LEFT JOIN history_events as HE on HE.identifier=EE.identifier '
+            'WHERE counterparty=? AND type=? AND subtype=?;', (
+                CPT_HOP,
+                HistoryEventType.RECEIVE.serialize(),
+                HistoryEventSubType.RECEIVE_WRAPPED.serialize(),
+            ),
+        ).fetchall()
+
+    progress_handler.set_total_steps(len(hop_lp_assets))
+
```

Comment:
the updates will happen fast so I wouldn't worry about setting the steps. Set just one step (update the protocol field in the assets).

Once that is done don't call in fetchall. Iterate over the cursor. This avoids iterating twice the result of the query